File: dehaze-python/app/models/entity/sys_dict.py
"""
字典相关实体模型
"""

from sqlalchemy import BigInteger, Index, Integer, String
from sqlalchemy.dialects import mysql as mysql_types
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysDict(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_dict"
    __table_args__ = {"comment": "字典数据表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    type_code: Mapped[str | None] = mapped_column(String(64), comment="字典类型编码")
    name: Mapped[str] = mapped_column(String(50), default="", comment="字典项名称")
    value: Mapped[str] = mapped_column(String(50), default="", comment="字典项值")
    sort: Mapped[int] = mapped_column(Integer, default=1, comment="排序")
    status: Mapped[int] = mapped_column(
        mysql_types.TINYINT, default=1, comment="状态(1:正常;0:禁用)"
    )
    defaulted: Mapped[int] = mapped_column(
        mysql_types.TINYINT, default=0, comment="是否默认(1:是;0:否)"
    )
    remark: Mapped[str] = mapped_column(String(255), default="", comment="备注")


class SysDictType(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_dict_type"
    __table_args__ = (Index("type_code", "code", unique=True), {"comment": "字典类型表"})

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键 "
    )
    name: Mapped[str] = mapped_column(String(50), default="", comment="类型名称")
    code: Mapped[str] = mapped_column(String(50), default="", comment="类型编码")
    status: Mapped[int] = mapped_column(
        mysql_types.TINYINT, default=1, comment="状态(1:正常;0:禁用)"
    )
    remark: Mapped[str | None] = mapped_column(String(255), comment="备注")

File: dehaze-go/internal/model/sys_dict.go
package model

// SysDict 字典数据表
type SysDict struct {
	BaseModel
	TypeCode  string `gorm:"column:type_code;type:varchar(64);comment:字典类型编码" json:"typeCode"`
	Name      string `gorm:"column:name;type:varchar(50);default:'';comment:字典项名称" json:"name"`
	Value     string `gorm:"column:value;type:varchar(50);default:'';comment:字典项值" json:"value"`
	Sort      int    `gorm:"column:sort;type:int;default:0;comment:排序" json:"sort"`
	Status    int8   `gorm:"column:status;type:tinyint;comment:状态(1:启用;0:禁用)" json:"status"`
	Defaulted int8   `gorm:"column:defaulted;type:tinyint;default:0;comment:是否默认(1:是;0:否)" json:"defaulted"`
	Remark    string `gorm:"column:remark;type:varchar(255);default:'';comment:备注" json:"remark"`
	Deleted   int8   `gorm:"column:deleted;type:tinyint;not null;default:0;comment:逻辑删除标识(0:未删除;1:已删除)" json:"deleted"`
}
